cmd/kube-proxy/app/server.go

```diff
@@ -628,7 +628,7 @@ func (s *ProxyServer) birthCry() {
 
 // (Note that on Linux, the node IPs are used only to determine whether a given
 // LoadBalancerSourceRanges value matches the node or not. In particular, they are *not*
-// used for NodePort handling.)
+// used for NodePort handling unless NodePortAddresses is not set to 'primary'.)
```

Comment:
The "not" is incorrect. They're used if NodePortAddresses *is* set to 'primary', not if it's *not* set to 'primary'.

But it would be better to rewrite the whole comment.

"(Note that on Linux, the node IPs are used only (a) to determine whether a given LoadBalancerSourceRanges value matches the node or not, and (b) as the IP(s) to accept NodePort connections on when NodePortAddresses is set to 'primary'.)"